homeassistant/components/lamarzocco/coordinator.py

```diff
@@ -80,6 +87,43 @@ async def _async_init_client(self) -> None:
                 name="lm_websocket_task",
             )
 
+        # initialize Bluetooth
+        if self.config_entry.data.get(CONF_USE_BLUETOOTH, True):
+
+            def bluetooth_configured() -> bool:
+                return self.config_entry.data.get(
+                    CONF_MAC, ""
+                ) and self.config_entry.data.get(CONF_NAME, "")
+
+            if not bluetooth_configured():
+                for discovery_info in async_discovered_service_info(self.hass):
+                    name = str(discovery_info.name)
+                    if (name.startswith(tuple(BT_MODEL_NAMES))) and (
```

Comment:
Move the objects that are being created here every loop outside the loop so you only create them once

Reply:
I don't get this one sorry. There is only a single thing updated and then the loop gets broken

Reply:
I pushed a suggestion

Reply:
Thank you. But I don't understand the benefit, sorry. What are we trying to achieve here?

Reply:
The goal is to not have to pay the cost of recreating the variables every loop since they are created outside the loop

Reply:
I'm not trying to argue with you here btw, just legitimately curious and trying to learn something, because for me it all looks the same... the only variable that is created is `name` and that still happens on every loop now

Reply:
`machine = self.config_entry.data[CONF_MACHINE]` is moved out side the loop

Let's unpack this one

`self.config_entry` is having to look up `config_entry` every loop .. similar to `getattr(self,'config_entry')`
`config_entry.data` is having to lookup `data` every loop .. similar to `getattr(config_entry,'data')`
`CONF_MACHINE` has to be looked up in `data` every loop

But assigning it to the local var `machine` it only has to lookup `machine`

`tuple(BT_MODEL_NAMES)` has to create the tuple every loop

You can disassemble the code with https://docs.python.org/3/library/dis.html to look at what its really doing

Reply:
tyvm. Understood, though I would have thought those to be ignorable performance wise (but I know you're the HA performance god, so I wanted to take that chance 😁). Why are you negating the condition though and using `continue`?

Reply:
> tyvm. Understood, though I would have thought those to be ignorable performance wise (but I know you're the HA performance god, so I wanted to take that chance 😁).

We have users with 2000+ visible bluetooth devices (or more) so you could end up creating 2000 tuples every time it looped on the miss case

> Why are you negating the condition though and using `continue`?

It makes the code more readable by saving indent

Reply:
kinda disagree on that last one, because the code itself gets harder to read imo (especially since that introduces an  uncovered line, which I needed to add an extra test for 😉)

Reply:
and btw. by using `and` we can avoid doing the `split` on every loop because we can jump out when the prefix doesn't match (can't we)

Reply:
> kinda disagree on that last one, because the code itself gets harder to read imo (especially since that introduces an uncovered line, which I needed to add an extra test for 😉)

Its just a suggestion, if you don't like it feel free to revert it and do it another way

Reply:
done. Sometimes it's hard to tell which reviews are suggestions and which are mandatory from the reviewers side.

Reply:
> and btw. by using `and` we can avoid doing the `split` on every loop because we can jump out when the prefix doesn't match (can't we)

As soon as one of the `or` is true the evaluation will stop, the`continue` will be reached so you'll never get to the split

Reply:
> especially since that introduces an uncovered line, which I needed to add an extra test for 😉

That's because the CI is measuring line cover, not branch coverage so the code is still uncovered even if you remove the guard.

Reply:
that's true of course, but tbh in this case I only care about line coverage, as there is no code to test for the other case

Reply:
> As soon as one of the `or` is true the evaluation will stop, the`continue` will be reached so you'll never get to the split
yeah, right ofc